src/deploy/functions/build.ts

```diff
@@ -224,12 +224,39 @@ export async function resolveBackend(
   const projectId = userEnvOpt.projectId;
   let paramValues: Record<string, Field<string | number | boolean>> = {};
   if (previews.functionsparams) {
-    paramValues = await params.resolveParams(build.params, projectId, userEnvs);
+    paramValues = await params.resolveParams(build.params, projectId, envWithTypes(userEnvs));
+
+    // TODO(vsfan@): when merging secrets support into the Build, make sure we aren't writing those to disk.
+    const toWrite: Record<string, string> = {};
+    for (const paramName of Object.keys(paramValues)) {
+      if (userEnvs.hasOwnProperty(paramName)) {
+        continue;
+      }
+      toWrite[paramName] = paramValues[paramName]?.toString() || "";
+    }
+    writeUserEnvs(toWrite, userEnvOpt);
   }
 
   return toBackend(build, paramValues);
 }
 
+function envWithTypes(rawEnvs: Record<string, string>): Record<string, string | number | boolean> {
```

Comment:
I'm nervous that casting this into a type now will cause problems. I worry that a customer has some reason to use "true" as a string. On first deploy it'll succeed because they'll be prompted for a string and provide `"true"`. But when it comes back, it'll come back as `true` and fail to update.

Reply:
Yeah, I'm worried about that too. I've been working on a branch where ParamValue is now a struct which contains both a raw actual value and some information about what types it can be applied to, but that system also interacts with Secret params so it doesn't make much sense to try and fit it into this branch. Call it a fast-follow sort of thing?